src/sidebar.ts

```diff
@@ -2,72 +2,79 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import {
+  SidebarCategoriesShorthand,
   SidebarItem,
   SidebarItemCategoryConfig,
+  SidebarItemConfig,
 } from "@docusaurus/plugin-content-docs/src/sidebars/types"
 
+type ExtendSidebar = SidebarItemConfig[]
+
 const homeLink: SidebarItem = {
   type: "link",
   href: "/welcome",
   className: "all-docs-link",
   label: "Go to Main Page",
 }
 
-const referenceSidebar = {
-  reference: [
-    homeLink,
-    {
-      type: "category",
-      label: "Reference",
-      collapsed: false,
-      collapsible: false,
-      items: [
-        "reference/api",
-        "sdk",
-        {
-          type: "category",
-          label: "Operations",
-          collapsed: false,
-          collapsible: false,
-          items: [
-            "concepts/personal-access-token",
-            "guides/cors",
-            "guides/rate-limits",
-            "guides/ip-allowlist",
-            "api/eventual-consistency",
-          ],
-        },
-        {
-          type: "category",
-          label: "Ory CLI",
-          collapsed: false,
-          collapsible: false,
-          items: [
-            "guides/cli/installation",
-            "guides/cli/cli-basics",
-            "guides/cli/identity-cli",
-            "guides/cli/proxy-and-tunnel",
-            "guides/cli/config-with-cli",
-            {
-              type: "category",
-              label: "CLI commands",
-              items: [
-                {
-                  type: "autogenerated",
-                  dirName: "cli",
-                },
-              ],
-            },
-          ],
-        },
-      ],
-    },
-  ],
-}
+const referenceSidebar: ExtendSidebar = [
+  homeLink,
+  {
+    type: "category",
+    label: "Reference",
+    collapsed: false,
+    collapsible: false,
+    items: [
+      "reference/api",
+      "sdk",
+      {
+        type: "category",
+        label: "Operations",
+        collapsed: false,
+        collapsible: false,
+        items: [
+          "concepts/personal-access-token",
+          "guides/cors",
+          "guides/rate-limits",
+          "guides/ip-allowlist",
+          "api/eventual-consistency",
+        ],
+      },
+      {
+        type: "category",
+        label: "Ory CLI",
+        collapsed: false,
+        collapsible: false,
+        items: [
+          "guides/cli/installation",
+          "guides/cli/cli-basics",
+          "guides/cli/identity-cli",
+          "guides/cli/proxy-and-tunnel",
+          "guides/cli/config-with-cli",
+          {
+            type: "category",
+            label: "CLI commands",
+            items: [
+              {
+                type: "autogenerated",
+                dirName: "cli",
+              },
+            ],
+          },
+        ],
+      },
+    ],
+  },
+]
+
+const quickstartSidebar = (flat: boolean): ExtendSidebar => {
```

Comment:
The idea is the following: the main documentation page shows the first 2 levels of the navigation (aka flat = true). If we browse on the main page (level 1), we see level 1 and 2 (intro, quickstarts, guides, ...). when we jump into one of the sub-items of a category (e.g. quickstarts), we jump to the next level where we now see level 2 and 3.

This way, we no longer need to keep in sync the main navigation and the "sub" navigation. but instead have one navigation tree that we can "go deeper" into.

Reply:
oh very cool, did not know this feature :D 

so basically the "flat" items we only show when you click on the category they are in, very nice

Reply:
Correct

Not really a future but a creative way to (ab)use the docusaurus nav bar :D